### ISSUE TITLE
Multidoc ask question show refernce link and highlight text on specific page issue - Wordpress/PHP

### ISSUE THREAD

**Issue by MacrewDev** [chatdoc-com/ChatDOC-API-Demo#17]
Hi Team,

I need your help to implement this functionality
https://chatdoc.com/chatdoc/#/upload

Here for multidoc highlight the text on specific page of upload id is
not working for me. Its showing wrong answers as reference link.

You can check here:
https://turtlestore.nl/chatdoc/
here I want to within all uploaded documents, its checking but as
answer is onlyp present in one doc but currently its showing many
references that is wrong.
https://turtlestore.nl/chat-test/

Also Its showing in canvas, I'm trying to fix it similar like your
reference. Can you please help me to fix out this issue? I really want
to fix it within a day.

Thanks

**Comment by Ronalci** [comment]
Hi MacrewDev！

1. Is https://turtlestore.nl/chatdoc/ your website? Could you please tell me how to log in or use it?
2. Could you provide a detailed explanation of the operating steps or how to use it? If possible, sharing your source code with us would also be very helpful.

**Comment by MacrewDev** [comment]
Hi Ronalci,

Thanks for reaching out to my query. Its little late but no problem. Hopefully we can solve it soon now.

This is not my website but its my client website.

**Comment by MacrewDev** [comment]
https://turtlestore.nl/register/
You can register from here.

**Comment by MacrewDev** [comment]
Let me give you my brief explanation first:
There was one other developer who implemented this feature but he is not here now. So I'm just trying to fix those issues which are not working correctly.

So I want below functionalities to work:
1) Single upload doc and multiple files or folder upload working Good - https://turtlestore.nl/chatdoc/
2) For multiple files, I want it should display the documents (pdf, docx, Images etc..)  as similar like your chatdoc. Even after asking question its not showing correct refernce link to highlight. Answer fetching is correct but reference link and highlight functionality is main issue here.
https://turtlestore.nl/chat-test/ 

3) For single doc, same issue of document view. So that highlight of specific text should work. And Refrence link is not implemented here for now. 
https://turtlestore.nl/chatdoc/chat/?upload_id=ad416f6e-8b6f-45fe-a953-1d44188e9c9f

I will share my source code as well. I know some of code is correctly added but many things are missed. I need your guidance to make this working perfectly.

Thanks

**Comment by MacrewDev** [comment]
[turtlestore-chatdoc-code.zip](https://github.com/user-attachments/files/16332007/turtlestore-chatdoc-code.zip)

Here I'm sharing my source code zip file. I using this in wordpress.
let me know if you wnat me to provide code in any other way.

**Comment by MacrewDev** [comment]
@Ronalci Can I know when I can get your response? I mean are you checking on this, Because I'm available for next 2 hours only. Then I will come by tomorrow 10am IST.

**Comment by Ronalci** [comment]
Received, I have passed the message on to my colleague, and she will reply to you as soon as possible.

**Comment by MacrewDev** [comment]
Hmm, let's see in how many days/months she will only reply. No expectation that they can help me to solve issue.

**Comment by Ronalci** [comment]
btw，are you looking to create a list for displaying multiple documents, or is there something else you would like to do?

**Comment by MacrewDev** [comment]
Yes, I'm working and my team also checking on that to solve this.

**Comment by Ronalci** [comment]
We do not provide a UI for a file list. Are you able to obtain the upload_id for multiple files, right? You can implement a list on the front end, and then fill this list with these upload_ids. Depending on your business needs, implement the functionality for multi-select or single-select files, and when asking a question, pass the selected upload_ids to ChatDOC.

**Comment by MacrewDev** [comment]
I want to show uploaded doc id in document previewer so that I can highlight the text from there. This the first thing that I'd like to resolve.

**Comment by Ronalci** [comment]
> Even after asking question its not showing correct refernce link to highlight

Regarding this issue, please study the code from the demo we provided. Alternatively, provide us with a JavaScript version of your source code.

**Comment by Ronalci** [comment]
> I want to show uploaded doc id in document previewer so that I can highlight the text from there. This the first thing that I'd like to resolve.

https://github.com/chatdoc-com/ChatDOC-API-Demo/blob/main/projects/client/src/views/CollectionChat.vue

**Comment by Ronalci** [comment]
> We do not provide a UI for a file list. Are you able to obtain the upload_id for multiple files, right? You can implement a list on the front end, and then fill this list with these upload_ids. Depending on your business needs, implement the functionality for multi-select or single-select files, and when asking a question, pass the selected upload_ids to ChatDOC.

https://github.com/chatdoc-com/ChatDOC-API-Demo/blob/main/projects/client/src/components/ChatView.vue

**Comment by MacrewDev** [comment]
Let me check for this.
